hnn_core/tests/test_optimize_evoked.py

```diff
@@ -0,0 +1,83 @@
+# Authors: Carolina Fernandez <cxf418@miami.edu>
+#          Nick Tolley <nicholas_tolley@brown.edu>
+#          Ryan Thorpe <ryan_thorpe@brown.edu>
+#          Mainak Jas <mjas@mgh.harvard.edu>
+
+from hnn_core import jones_2009_model, simulate_dipole, Optimizer
+
+
+def _optimize_evoked(solver):
+    """Test running the full routine in a reduced network."""
+
+    tstop = 10.
+    n_trials = 1
+
+    # simulate a dipole to establish ground-truth drive parameters
+    net_orig = jones_2009_model()
+    net_orig._N_pyr_x = 3
+    net_orig._N_pyr_y = 3
+    mu_orig = 2.
+    weights_ampa = {'L2_basket': 0.5,
+                    'L2_pyramidal': 0.5,
+                    'L5_basket': 0.5,
+                    'L5_pyramidal': 0.5}
+    synaptic_delays = {'L2_basket': 0.1, 'L2_pyramidal': 0.1,
+                       'L5_basket': 1., 'L5_pyramidal': 1.}
+    net_orig.add_evoked_drive('evprox',
+                              mu=mu_orig,
+                              sigma=1,
+                              numspikes=1,
+                              location='proximal',
+                              weights_ampa=weights_ampa,
+                              synaptic_delays=synaptic_delays)
+    dpl_orig = simulate_dipole(net_orig, tstop=tstop, n_trials=n_trials)[0]
+
+    # define set_params function and constraints
+    net_offset = jones_2009_model()
+    net_offset._N_pyr_x = 3
+    net_offset._N_pyr_y = 3
+
+    def set_params(net_offset, params):
+        weights_ampa = {'L2_basket': 0.5,
+                        'L2_pyramidal': 0.5,
+                        'L5_basket': 0.5,
+                        'L5_pyramidal': 0.5}
+        synaptic_delays = {'L2_basket': 0.1, 'L2_pyramidal': 0.1,
+                           'L5_basket': 1., 'L5_pyramidal': 1.}
+        net_offset.add_evoked_drive('evprox',
+                                    mu=params['mu_offset'],
+                                    sigma=1,
+                                    numspikes=1,
+                                    location='proximal',
+                                    weights_ampa=weights_ampa,
+                                    synaptic_delays=synaptic_delays)
+
+    # define constraints
+    mu_range = (1, 6)
+    constraints = dict()
+    constraints.update({'mu_offset': mu_range})
```

Comment:
Has `mu` actually been offset, or is it just the name of the "mu" that is getting optimized?

Reply:
Good catch, no, it hasn't been offset. The key should just be "mu", I just corrected it.